Lesson21/todo-app/sahra/index.html

```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>To-Do App</title>
+    <link rel="stylesheet" type="text/css" href="styles.css" />
+  </head>
+  <body>
+    <main class="container">
+      <form class="todo-form" id="todoForm">
+        <h1><label for="todoInput">To-Do List</label></h1>
+        <input placeholder="Add to do item" id="todoInput" class="todo-form-item"/>
+        <button type="submit" onclick="addItem()">ADD</button>
```

Comment:
You can execute JS code by adding onclick attribute to the buttons and other elements in HTML, however it's not a preferred approach.

**Using event listeners in JS instead of onclick attributes in HTML is preferred because:**

- It separates structure (HTML) from behavior (JS), improving maintainability
- Allows attaching multiple handlers to one event
- Provides more control over event handling (e.g., propagation, removal)
- Centralizes event logic, making code easier to manage
- Enables event delegation for better performance
- Follows modern web development best practices